nikola/plugins/task/archive.py

```diff
@@ -150,10 +152,34 @@ def provide_context_and_uptodate(self, classification, lang, node=None):
             kw["is_feed_stale"] = (datetime.datetime.utcnow().strftime("%Y/%m/%d") != classification)
         else:
             raise Exception("Cannot interpret classification {}!".format(repr(classification)))
+
         context = {
             "title": title,
             "pagekind": [page_kind, "archive_page"],
         }
+
+        # Generate links for hierarchies
+        if kw["create_archive_navigation"]:
+            if hierarchy:
+                # Up level link makes sense only if this is not the top-level
+                # page (hierarchy is empty)
+                context["up_level"] = self.site.link('/'.join(hierarchy[:-1]), lang)
+            else:
+                context["up_level"] = None
+
+            flat_hierarchy = self.site.flat_hierarchy_per_classification['archive'][lang]
+            nodelevel = len(hierarchy)
```

Comment:
You should also insert `nodelevel` into the context so that link texts (next year/month/day) can be chosen appropriately without unnecessary complicated code (like manually counting the number of slashes in the classification etc.).